File: python/binary_tree_level_order_traversal.py
# no.102
from utils import TreeNode

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        ans = []
        if not root:
            return ans
        queue = [root]
        while queue:
            row = []
            next_row = []
            for node in queue:
                row.append(node.val)
                if node.left:
                    next_row.append(node.left)
                if node.right:
                    next_row.append(node.right)
            ans.append(row)
            queue = next_row
        return ans

File: C++/binary_tree_level_order_traversal.cpp
// no.102
#include "tree_node.h"
#include <vector>
using namespace std;

class Solution {
public:
  vector<vector<int>> levelOrder(TreeNode *root) {
    vector<vector<int>> ans;
    if (!root) {
      return ans;
    }
    vector<int> row;
    vector<TreeNode *> next_row;
    vector<TreeNode *> queue;
    queue.push_back(root);
    while (!queue.empty()) {
      row.clear();
      next_row.clear();
      for (TreeNode *node : queue) {
        row.push_back(node->val);
        if (node->left) {
          next_row.push_back(node->left);
        }
        if (node->right) {
          next_row.push_back(node->right);
        }
      }
      ans.push_back(row);
      queue = next_row;
    }
    return ans;
  }
};
